### PR TITLE
Support SASL plain authentication

### PR DESCRIPTION
**Pull request by gabrielgiordan**
Supports SASL plain mechanism authentication to connect to Confluent Cloud Schema Registry

### CODE REVIEW

File: src/avlizer.erl [hunk 1]
```diff
@@ -25,13 +25,25 @@
...
-  case osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_URL", false) of
-    false -> ok;
-    URL ->
-      Vars0 = application:get_env(?APPLICATION, avlizer_confluent, #{}),
-      Vars = Vars0#{schema_registry_url => URL},
-      application:set_env(?APPLICATION, avlizer_confluent, Vars)
+  SchemaRegistryURL = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_URL", false),
+  SchemaRegistrySASLMechanism = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_SASL_MECHANISM", false),
+  SchemaRegistrySASLUsername = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_SASL_USERNAME", false),
+  SchemaRegistrySASLPassword = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_SASL_PASSWORD", false),
```
Comment (k32) [inline src/avlizer.erl]: It's a good practice to wrap passwords in a closure, so they don't end up in the crash reports:

```
 SchemaRegistrySASLPassword0 = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_SASL_PASSWORD", false),
 SchemaRegistrySASLPassword = fun() ->SchemaRegistrySASLPassword0 end,
 ...
```

File: src/avlizer_confluent.erl [hunk 2]
```diff
@@ -303,10 +303,17 @@ terminate(_Reason, _State) ->
 
 %%%_* Internals ================================================================
 
-get_registry_url() ->
-  {ok, #{ schema_registry_url := URL
-        }} = application:get_env(?APPLICATION, ?MODULE),
-  URL.
+get_registry_base_request() ->
+  case application:get_env(?APPLICATION, ?MODULE) of
+    {ok, #{schema_registry_url := URL, schema_registry_sasl := SASL}} ->
```
Comment (k32) [inline src/avlizer_confluent.erl]: sys.conf is not the best place to store the secrets. Please consider supporting retrieving this data from a SASL file.

**Comment by k32** [comment]
Hey! Thanks for the PR. Could you please look through the remarks and merge conflicts.

**Comment by gabrielgiordan** [comment]
> Hey! Thanks for the PR. Could you please look through the remarks and merge conflicts. Also it would be nice to add a testcase for this.

Hello @k32 ! Thank you for the review, I've implemented all your suggestions since I agree with it all.

**Pull request closed without merging**